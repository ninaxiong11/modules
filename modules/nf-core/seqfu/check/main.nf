process SEQFU_CHECK {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/seqfu:1.22.3--hc29b5fc_1':
        'biocontainers/seqfu:1.22.3--hc29b5fc_1' }"

    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("${prefix}.tsv"), emit: check
    path "versions.yml"                   , emit: versions

    when:                                  
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"

    def dirFlag = (reads instanceof List ? reads.every { it.isDirectory() } : reads.isDirectory()) ? "--dir" : ""

    """
    seqfu \\
        check \\
        ${args} \\
        ${dirFlag} ${reads} > ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqfu: \$(seqfu --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo $args
    
    touch ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqfu: \$(seqfu --version)
    END_VERSIONS
    """
}
