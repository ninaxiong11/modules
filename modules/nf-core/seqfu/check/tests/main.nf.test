nextflow_process {

    name "Test Process SEQFU_CHECK"
    script "../main.nf"
    process "SEQFU_CHECK"

    tag "modules"
    tag "modules_nfcore"
    tag "seqfu"
    tag "seqfu/check"

    test("sarscov2 - fastq - single-end") {

        when {
            process {
                """
                input[0] = [[ id:'test', single_end:true ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - fastq - paired-end") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test_paired', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - fastq - directory") {
        when {
            process {
                """
                // stage input fastq files in a folder
                tmpdir = file("tmpdir", type: 'dir')
                tmpdir.mkdir()
                fastq_file_1 = file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                fastq_file_2 = file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                fastq_file_1.copyTo(tmpdir)
                fastq_file_2.copyTo(tmpdir)

                // define inputs of the process here. Example:
                input[0] = [
                    [ id:'test' ],
                    [
                        file(tmpdir, type:'dir', checkIfExists: true)
                    ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    path(process.out.versions.get(0)).yaml
                ).match() }
            )
        }
    }
 
     test("sarscov2 - fastq - single_end - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [[ id:'test', single_end:true ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    process.out.versions.collect{ path(it).yaml }.unique()
                ).match() }
            )
        }

    }

}