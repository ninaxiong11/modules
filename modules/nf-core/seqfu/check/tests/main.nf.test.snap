{
    "sarscov2 - fastq - single_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,68acb41f0d0333843fd5d1b0c7256fd1"
                ],
                "check": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68acb41f0d0333843fd5d1b0c7256fd1"
                ]
            },
            [
                {
                    "SEQFU_CHECK": {
                        "seqfu": "1.22.3"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-29T12:44:11.259028502"
    },
    "sarscov2 - fastq - directory": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tsv:md5,cd149b0f08b698981591be4a25729d7e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,68acb41f0d0333843fd5d1b0c7256fd1"
                ],
                "check": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tsv:md5,cd149b0f08b698981591be4a25729d7e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68acb41f0d0333843fd5d1b0c7256fd1"
                ]
            },
            {
                "SEQFU_CHECK": {
                    "seqfu": "1.22.3"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-29T12:43:55.867776056"
    },
    "sarscov2 - fastq - paired-end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_paired",
                            "single_end": false
                        },
                        "test_paired.tsv:md5,c3e6eae028f23041bd09cac9c371f318"
                    ]
                ],
                "1": [
                    "versions.yml:md5,68acb41f0d0333843fd5d1b0c7256fd1"
                ],
                "check": [
                    [
                        {
                            "id": "test_paired",
                            "single_end": false
                        },
                        "test_paired.tsv:md5,c3e6eae028f23041bd09cac9c371f318"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68acb41f0d0333843fd5d1b0c7256fd1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-29T12:43:35.832380663"
    },
    "sarscov2 - fastq - single-end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.tsv:md5,4c6409169772005cfb06be9e41f2c1e2"
                    ]
                ],
                "1": [
                    "versions.yml:md5,68acb41f0d0333843fd5d1b0c7256fd1"
                ],
                "check": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.tsv:md5,4c6409169772005cfb06be9e41f2c1e2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68acb41f0d0333843fd5d1b0c7256fd1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-29T12:43:16.970955378"
    }
}